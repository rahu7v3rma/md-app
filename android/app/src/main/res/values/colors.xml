<?xml version="1.0" encoding="utf-8"?>
<resources>
    <color name="primary">#72996B</color>
    <color name="splash">#85B085</color>
</resources>
