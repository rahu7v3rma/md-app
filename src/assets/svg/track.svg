<svg width="18" height="21" viewBox="0 0 18 21" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M11.4513 18.621C9.9292 19.477 8.0708 19.477 6.5487 18.621L3.5487 16.934C1.9743 16.048 1 14.382 1 12.576V9.424C1 7.618 1.9743 5.952 3.5487 5.066L6.5487 3.379C8.0708 2.523 9.9292 2.523 11.4513 3.379L14.4513 5.066C16.0257 5.952 17 7.618 17 9.424V11" stroke="#FF9330" stroke-width="1.5" stroke-linecap="round"/>
  <path fill-rule="evenodd" clip-rule="evenodd" d="M5.931 2.738C7.6814 1.754 9.8186 1.754 11.569 2.738L14.569 4.42599C16.3795 5.44399 17.5 7.36 17.5 9.438V11.013C17.5 11.356 17.2698 11.645 16.9556 11.735C16.9484 11.745 16.941 11.754 16.9334 11.763C16.7306 11.763 16.5159 11.765 16.2775 11.768C15.7127 11.773 15.0155 11.779 14.0312 11.763C12.7007 11.741 12.012 12.183 12.012 13.513V17.21C12.012 17.939 12.0037 18.393 11.9499 18.681C11.9343 18.926 11.7989 19.159 11.569 19.288C9.8186 20.273 7.6814 20.273 5.931 19.288L2.931 17.601C1.1205 16.582 0 14.666 0 12.589V9.43699C0 7.35999 1.1205 5.44399 2.931 4.42599L5.931 2.738ZM9 6.24999C9.4142 6.24999 9.75 6.58599 9.75 6.99999V10.701C10.1984 10.96 10.5 11.445 10.5 12C10.5 12.828 9.8284 13.5 9 13.5C8.1716 13.5 7.5 12.828 7.5 12C7.5 11.445 7.8016 10.96 8.25 10.701V6.99999C8.25 6.58599 8.5858 6.24999 9 6.24999Z" fill="#FF9330"/>
  <path d="M3 3L6 1" stroke="#FF9330" stroke-width="1.5" stroke-linecap="round"/>
  <path d="M15 3L12 1" stroke="#FF9330" stroke-width="1.5" stroke-linecap="round"/>
  <path d="M13 14.996V14.16C13 13.128 13.7861 12.702 14.7418 13.222L15.5125 13.64L16.2832 14.059C17.2389 14.578 17.2389 15.422 16.2832 15.941L15.5125 16.36L14.7418 16.778C13.7861 17.298 13 16.872 13 15.84V14.996Z" stroke="#FF9330" stroke-width="1.5" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
