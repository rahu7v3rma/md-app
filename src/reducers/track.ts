import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
import moment from 'moment';
import { useSelector } from 'react-redux';
import { PURGE } from 'redux-persist';

import { registerTriggerNotifications } from '@/services/notification';
import { RootState } from '@/store';
import { Constants } from '@/utils/constants';

interface TrackState {
    loading: boolean;
    trackFastState: '' | 'tracking' | 'tracked' | 'complete';
    trackStartedAt: string;
    trackedTimeInSeconds: number;
    trackTimeLimitInSeconds: number;
    trackStartTimeInSecond: number;
}

const initialState: TrackState = {
    loading: false,
    trackFastState: '',
    trackStartedAt: '',
    trackedTimeInSeconds: 0,
    trackTimeLimitInSeconds: 16 * (60 * 60),
    trackStartTimeInSecond: 16 * (60 * 60)
};

export const trackStart = createAsyncThunk(
    'track/start',
    async (options: { startDateTime: string }, { rejectWithValue }) => {
        try {
            return options;
        } catch (err) {
            return rejectWithValue(err);
        }
    }
);

export const trackedTimeUpdateAuto = createAsyncThunk(
    'track/time/update/auto',
    () => true
);

export const trackedTimeUpdateManual = createAsyncThunk(
    'track/time/update/manual',
    async (options: { trackedTimeInSeconds: number }) => {
        return options;
    }
);

export const trackUpdate = createAsyncThunk(
    'track/update',
    async (options?: { timeToUpdate: number }) => {
        return options;
    }
);

export const trackTimeLimitUpdate = createAsyncThunk(
    'track/updateTime',
    async (options: { timeLimitInSeconds: number }) => {
        return options;
    }
);
export const trackStartTime = createAsyncThunk(
    'start/track',
    async (options: { startTimeInSecond: number }) => {
        return options;
    }
);

export const trackStop = createAsyncThunk('track/stop', async () => {
    return true;
});

export const trackReset = createAsyncThunk('track/reset', async () => {
    return true;
});

export const trackComplete = createAsyncThunk('track/complete', async () => {
    const date = moment().add(2, 'second').toString();
    registerTriggerNotifications(
        new Date(date),
        'Fast is completed',
        'Your fast is completed please log your fast.',
        Constants.notificationType.FAST_COMPLETE
    );
    return true;
});

export const timerStartEnd = createAsyncThunk(
    'track/timer/startEnd',
    async (options: { timerStartedAt: string; timerEndAt: string }) => {
        return options;
    }
);

const trackSlice = createSlice({
    name: 'track',
    initialState,
    reducers: {},
    extraReducers: (builder) => {
        builder.addCase(trackStart.pending, (state) => {
            state.loading = true;
        });
        builder.addCase(trackStart.fulfilled, (state, action) => {
            state.loading = false;
            state.trackFastState = 'tracking';
            state.trackStartedAt = action.payload.startDateTime;
        });
        builder.addCase(trackStart.rejected, (state) => {
            state.loading = false;
        });

        builder.addCase(trackedTimeUpdateAuto.fulfilled, (state) => {
            state.loading = false;
            state.trackedTimeInSeconds += 1;
        });
        builder.addCase(trackedTimeUpdateManual.fulfilled, (state, action) => {
            state.loading = false;
            state.trackedTimeInSeconds = action.payload.trackedTimeInSeconds;
        });

        builder.addCase(trackUpdate.fulfilled, (state, action) => {
            state.loading = false;
            if (action.payload && action.payload.timeToUpdate) {
                state.trackedTimeInSeconds =
                    state.trackedTimeInSeconds + action.payload.timeToUpdate;
            } else {
                state.trackedTimeInSeconds = state.trackedTimeInSeconds + 1;
            }
        });

        builder.addCase(trackTimeLimitUpdate.fulfilled, (state, action) => {
            state.trackTimeLimitInSeconds =
                action.payload?.timeLimitInSeconds ?? 16 * (60 * 60);
        });
        builder.addCase(trackStartTime.fulfilled, (state, action) => {
            state.trackStartTimeInSecond =
                action.payload?.startTimeInSecond ?? 16 * (60 * 60);
        });

        builder.addCase(trackStop.fulfilled, (state) => {
            state.loading = false;
            state.trackFastState = 'tracked';
        });

        builder.addCase(trackComplete.fulfilled, (state) => {
            state.loading = false;
            state.trackFastState = 'complete';
        });

        builder.addCase(trackReset.fulfilled, (state) => {
            state.loading = false;
            state.trackFastState = '';
            state.trackStartedAt = '';
            state.trackedTimeInSeconds = 0;
            state.trackTimeLimitInSeconds = 16 * (60 * 60);
            state.trackStartTimeInSecond = 16 * (60 * 60);
        });

        // when purging reset back to the initial state
        builder.addCase(PURGE, () => initialState);
    }
});

export default trackSlice.reducer;

interface TrackSelectorsType {
    loading: boolean;
    trackFastState: '' | 'tracking' | 'tracked' | 'complete';
    trackStartedAt: string;
    trackedTimeInSeconds: number;
    trackTimeLimitInSeconds: number;
    trackStartTimeInSecond: number;
}

export const TrackSelectors = (): TrackSelectorsType => {
    const loading = useSelector((state: RootState) => state.track.loading);

    const trackStartedAt = useSelector(
        (state: RootState) => state.track.trackStartedAt
    );
    const trackFastState = useSelector(
        (state: RootState) => state.track.trackFastState
    );

    const trackedTimeInSeconds = useSelector(
        (state: RootState) => state.track.trackedTimeInSeconds
    );

    const trackTimeLimitInSeconds = useSelector(
        (state: RootState) => state.track.trackTimeLimitInSeconds
    );
    const trackStartTimeInSecond = useSelector(
        (state: RootState) => state.track.trackStartTimeInSecond
    );

    return {
        loading,
        trackFastState,
        trackStartedAt,
        trackedTimeInSeconds,
        trackTimeLimitInSeconds,
        trackStartTimeInSecond
    };
};
