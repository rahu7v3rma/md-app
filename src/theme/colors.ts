// theme colors
const PRIMARY = '#72996B';
const PRIMARY_LIGHT = '#D3E29F40';
const BACKGROUND = '#F5F5F5';
const SHEET_BACKGROUND_COLOR = 'rgba(0,0,0,0.5)';
const BACKGROUND_LIGHTEST = '#FAFAFF';
const LOG_PAGE_BACKGROUND_COLOR = '#f5f4ee';

// INPUT
const INPUT = '#D3E6F8';
const INPUT_BG = '#EEF4FA';
const INPUT_ERROR_BG = '#FFECEC';
const INPUT_ERROR_BORDER = '#F9D4D4';
const INPUT_ERROR_PLACEHOLDER = '#D05151';
const INPUT_BORDER = '#D3E29F';

// PROGRESS BAR
const PROGRESS_BAR_FILL = '#4DBA0B';
const PROGRESS_BAR_FILL_BORDER = '#3A9A00';
const PROGRESS_BAR_ACTIVITY_INDICATOR = PRIMARY;
const PROGRESS_BAR_FILL_GREEN = '#72996B';
const PROGRESS_BAR_FILL_DISABLE = '#BFC4C9';
const PROGRESS_BAR_TRACK_GREEN = '#EDF0DA';

// PROGRESS CIRCLE
const PROGRESS_CIRCLE_COLOR = '#f00';
const PROGRESS_CIRCLE_SHADOW_COLOR = '#999';
const PROGRESS_CIRCLE_BG_COLOR = '#e9e9ef';
const PROGRESS_CIRCLE_START_BALL_COLOR = '#72996B';
const PROGRESS_CIRCLE_START_BALL_DISABLE_COLOR = '#BDBECA';
const PROGRESS_CIRCLE_INNER_BALL_COLOR = '#5B37BE';
const PROGRESS_CIRCLE_INNER_BALL_DISABLE_COLOR = '#A4AAAF';

// SLIDER
const SLIDER_BG = 'rgba(120, 120, 128, 0.32)';

// BUTTON
const BUTTON_BORDER = '#D3E6F8';
const GREEN_CHECK_DARK = '#D3E6F8';
const BUTTON_LIGHTER_BORDER = '#f4f8e7';
const BUTTON_LIGHTER_BACKGROUND = '#f4f8e7';
const BUTTON_PRIMARY_BORDER = '#5A3DBF';
const BUTTON_DISABLED_BG = '#cccccc';
const BUTTON_DISABLED_BORDER = '#D6D6D6';
const BUTTON_IMAGE_BACKGROUND = '#F0E957';
const BUTTON_ICON_BACKGROUND = '#D3E29F40';
const BUTTON_RED_BACKGROUND = '#B16663';
const BUTTON_SUCCESS_BACKGROUND = '#4DBA0B';
const BUTTON_GREEN_BACKGROUND = '#72996B';
const BUTTON_WHITE_BACKGROUND = '#FFFFFF';
const BUTTON_DARK_RED_BACKGROUND = '#CD4049';

const TEXT_BLACK = '#271A51'; // more of a dark-blue
const TEXT_DARK_BLACK = '#1F1E22';
const TEXT_BLACK_GRAY = '#4D4F4D';
const TEXT_GRAY_BLACK = '#4D4F4D';
const TEXT_WHITE = '#FFFFFF';
const TEXT_GREEN = '#72996B';
const TEXT_CHARCOAL = '#4D4F4D';
const TEXT_DUSKY_OLIVE = '#847F60';
const TEXT_ERROR = '#D05151';
const TEXT_LIGHTER = 'rgba(235, 235, 245, 0.6)';
const TEXT_DISABLED = '#D6D6D6';
const TEXT_GRAY = '#A4AAAF';
const TEXT_GRAY_BASE = '#58595A';
const TEXT_PRIMARY = PRIMARY;
const TEXT_PURPLE = '#6E51D0';
const TEXT_LINK = '#3ED1FF';
const TEXT_DARK_GREEN = '#72996B';
const TEXT_DARKER = '#4D4F4D';
const TEXT_MAIN_DARKER = '#4D4F4D';

//ICONS
const EYE_ICON = '#6E51D0';
const EYE_ICON_GREEN = '#72996B';
const EYE_ICON_ERROR_BORDER = '#D05151';
const GREEN_CHECK_MARK = '#4DBA0B';

// toast messages
const ERROR_BACKGROUND = '#D05151';
const SUCCESS_BACKGROUND = '#4DBA0B';
const INFO_BACKGROUND = '#A4AAAF';

//Modals
const MODAL_BACKDROP = 'rgba(0,0,0,0.5)';
const MODAL_CLOSE_HANDLE = '#F5F8FB';
const MODAL_ITEM_DIVIDER = '#E6E8EC';

//EXTRAS
const SUCCESS_LIGHTEST = '#ECF8E6';
const SUCCESS_LIGHTER = '#DFEED8';
const DARK_CHARCOAL = '#1F1E22';
const WHITE = '#FFFFFF';
const BLACK = '#000000';
const BLACK_LIGHTER = 'rgba(37, 37, 37, 0.6)';
const JOURNEY_BACKGROUND = '#CEE1FE';
const JOURNEY_SECONDARY = '#ADCAF4';
const BLOCK_IN_PROGRESS = PRIMARY;
//const BLOCK_COMPLETE = '#4DBA0B';
const ERROR_LIGHTEST = '#FFECEC';
const ERROR_LIGHTER = '#F9D4D4';
const ERROR_BASE = '#D05151';
const BLACK_SHADOW = '#171717';
const CARD_COMPLETED = '#eff8e7';
const CARD_INPROGRESS = '#f0f4f8';
const PAGE_BG = '#F5F8FB';
const LIGHT_BACK_COLOR = '#D3E29F40';
const BLACK_CARDS = 'rgba(0, 0, 0, 0.3)';
const HEADER_SHADOW = 'rgba(0, 0, 0, 0.03)';
const GRAY_SHADOW = '#0000000D';
const BORDER_COLOR = '#E1E6C2';
const TRANSPARENT_COLOR = 'transparent';
const BADGE_RED = '#F50157';

//SWITCH
const SWITCH_TOGGLED = '#72996B';
const SWITCH_BUTTON = WHITE;

// pagination
const ACTIVE_PAGINATION_DOT = PRIMARY;
const INACTIVE_PAGINATION_DOT = PRIMARY_LIGHT;

export const Colors = {
    theme: {
        primary: PRIMARY,
        primary_light: PRIMARY_LIGHT,
        app_background: BACKGROUND,
        app_sheet_background_color: SHEET_BACKGROUND_COLOR,
        app_background_lightest: BACKGROUND_LIGHTEST,
        log_page_background_color: LOG_PAGE_BACKGROUND_COLOR
    },
    text: {
        black: TEXT_BLACK,
        black_gray: TEXT_BLACK_GRAY,
        dark_green: TEXT_DARK_GREEN,
        text_gray_black: TEXT_GRAY_BLACK,
        dark_black: TEXT_DARK_BLACK,
        white: TEXT_WHITE,
        error: TEXT_ERROR,
        green: TEXT_GREEN,
        charcoal: TEXT_CHARCOAL,
        dusky_olive: TEXT_DUSKY_OLIVE,
        text_lighter: TEXT_LIGHTER,
        text_disabled: TEXT_DISABLED,
        gray: TEXT_GRAY,
        gray_Base: TEXT_GRAY_BASE,
        primary: TEXT_PRIMARY,
        purple: TEXT_PURPLE,
        link: TEXT_LINK,
        darker: TEXT_DARKER,
        mainDarker: TEXT_MAIN_DARKER
    },
    input: {
        error_bg: INPUT_ERROR_BG,
        error_border: INPUT_ERROR_BORDER,
        app_input: INPUT,
        app_input_bg: INPUT_BG,
        input_placeholder_error: INPUT_ERROR_PLACEHOLDER,
        input_border: INPUT_BORDER,
        light_back_color: LIGHT_BACK_COLOR
    },
    button: {
        app_button_border: BUTTON_BORDER,
        app_button_lighter_background: BUTTON_LIGHTER_BACKGROUND,
        app_button_lighter_border: BUTTON_LIGHTER_BORDER,
        app_button_primary_border: BUTTON_PRIMARY_BORDER,
        app_button_disabled_bg: BUTTON_DISABLED_BG,
        app_button_disabled_border: BUTTON_DISABLED_BORDER,
        app_button_background: BUTTON_IMAGE_BACKGROUND,
        app_icon_button_background: BUTTON_ICON_BACKGROUND,
        app_button_red_background: BUTTON_RED_BACKGROUND,
        app_button_success_background: BUTTON_SUCCESS_BACKGROUND,
        app_button_green_background: BUTTON_GREEN_BACKGROUND,
        app_button_white_background: BUTTON_WHITE_BACKGROUND,
        app_btn_light_color: LIGHT_BACK_COLOR,
        app_button_dark_red_background: BUTTON_DARK_RED_BACKGROUND
    },
    icons: {
        eye: EYE_ICON,
        eyeGreen: EYE_ICON_GREEN,
        eye_error_border: EYE_ICON_ERROR_BORDER,
        green_check_mark: GREEN_CHECK_MARK,
        green_dark_check: GREEN_CHECK_DARK
    },
    modals: {
        backdrop: MODAL_BACKDROP,
        close_handle: MODAL_CLOSE_HANDLE,
        divider: MODAL_ITEM_DIVIDER
    },
    toast: {
        errorBackground: ERROR_BACKGROUND,
        successBackground: SUCCESS_BACKGROUND,
        infoBackground: INFO_BACKGROUND
    },
    progress: {
        fillWhite: TEXT_WHITE,
        fill: PROGRESS_BAR_FILL,
        fill_border: PROGRESS_BAR_FILL_BORDER,
        activity_indicator: PROGRESS_BAR_ACTIVITY_INDICATOR,
        fillGreen: PROGRESS_BAR_FILL_GREEN,
        fillDisable: PROGRESS_BAR_FILL_DISABLE,
        trackColor: PROGRESS_BAR_TRACK_GREEN,
        headerProgressColor: PROGRESS_CIRCLE_INNER_BALL_COLOR
    },
    slider: {
        bg: SLIDER_BG
    },
    extras: {
        success_lightest: SUCCESS_LIGHTEST,
        success_lighter: SUCCESS_LIGHTER,
        white: WHITE,
        black: BLACK,
        black_lighter: BLACK_LIGHTER,
        blue_background: JOURNEY_BACKGROUND,
        blue_secondary: JOURNEY_SECONDARY,
        block_in_progress: BLOCK_IN_PROGRESS,
        block_complete: BLOCK_IN_PROGRESS,
        error_lighter: ERROR_LIGHTER,
        error_lightest: ERROR_LIGHTEST,
        error_base: ERROR_BASE,
        black_shadow: BLACK_SHADOW,
        card_completed: CARD_COMPLETED,
        card_inProgress: CARD_INPROGRESS,
        page_bg: PAGE_BG,
        blackCards: BLACK_CARDS,
        headerShadow: HEADER_SHADOW,
        dark_charcoal: DARK_CHARCOAL,
        block_locked: INPUT_BORDER,
        gray_shadow: GRAY_SHADOW,
        border_color: BORDER_COLOR,
        transparent: TRANSPARENT_COLOR,
        badgeRed: BADGE_RED
    },
    pagination: {
        active_dot: ACTIVE_PAGINATION_DOT,
        inactive_dot: INACTIVE_PAGINATION_DOT
    },
    progressCircle: {
        color: PROGRESS_CIRCLE_COLOR,
        shadow_color: PROGRESS_CIRCLE_SHADOW_COLOR,
        bg_color: PROGRESS_CIRCLE_BG_COLOR,
        start_ball_color: PROGRESS_CIRCLE_START_BALL_COLOR,
        start_ball_disable_color: PROGRESS_CIRCLE_START_BALL_DISABLE_COLOR,
        inner_ball_color: PRIMARY,
        inner_ball_disable_color: PROGRESS_CIRCLE_INNER_BALL_DISABLE_COLOR
    },
    switch: {
        button: SWITCH_BUTTON,
        toggled: SWITCH_TOGGLED
    },
    statusBar: {
        white: WHITE
    }
};
