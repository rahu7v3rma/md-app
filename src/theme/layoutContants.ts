export const LayoutContants = {
    gapSpacing: 5,
    cardHeight: 90
};
