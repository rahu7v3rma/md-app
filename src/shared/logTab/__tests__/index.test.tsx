import React from 'react';
import { StyleSheet } from 'react-native';
import renderer, { act } from 'react-test-renderer';

import track from '@/assets/svg/track.svg';
import LogTab from '@/shared/logTab';
import { Colors } from '@/theme/colors';
import { LayoutContants } from '@/theme/layoutContants';

const logTabStyle = StyleSheet.create({
    wrapper: {
        flexDirection: 'row',
        alignItems: 'center',
        height: LayoutContants.cardHeight,
        width: '44%',
        borderWidth: 1.5,
        backgroundColor: Colors.extras.white,
        borderRadius: 20,
        shadowColor: Colors.theme.app_sheet_background_color,
        elevation: 3
    }
});

it('logTab snapshot', async () => {
    const tree = renderer.create(<LogTab icon={track} title="Fasting Timer" />);
    expect(tree).toMatchSnapshot();
});

it('logTab renders without crashing', async () => {
    const tree = renderer.create(<LogTab icon={track} title="Fasting Timer" />);
    expect(tree.root).toBeTruthy();
});

it('displays title and icon correctly', async () => {
    const tree = renderer.create(<LogTab icon={track} title="Fasting Timer" />);
    const logTabTitle = tree.root.findByProps({
        testID: 'logTabTitle'
    }).props;
    expect(logTabTitle.children).toBe('Fasting Timer');

    const logTabIcon = tree.root.findByProps({
        testID: 'logTabIcon'
    }).props;
    expect(logTabIcon.xml).toBe(track);
});

it('applies styles correctly', async () => {
    const active = false;
    const logTabContainerStyle = {
        width: '90%'
    };
    const logTabTextStyle = {
        width: '100%'
    };
    const tree = renderer.create(
        <LogTab
            icon={track}
            title="Fasting Timer"
            active={active}
            style={logTabContainerStyle}
            textStyle={logTabTextStyle}
        />
    );

    const logTabContainer = tree.root.findByProps({
        testID: 'logTab'
    }).props;
    expect(logTabContainer.style).toStrictEqual([
        logTabStyle.wrapper,
        {
            borderColor: active ? Colors.theme.primary : Colors.extras.white
        },
        logTabContainerStyle
    ]);

    const logTabTitleContainer = tree.root.findByProps({
        testID: 'logTabTitleContainer'
    }).props;
    expect(logTabTitleContainer.style).toStrictEqual(logTabTextStyle);
});

it('calls onTabPress when tab is pressed', async () => {
    const onTabPress = jest.fn();
    const tree = renderer.create(
        <LogTab icon={track} title="Fasting Timer" onTabPress={onTabPress} />
    );
    const logTabContainer = tree.root.findByProps({
        testID: 'logTab'
    }).props;
    await act(() => logTabContainer.onPress());
    expect(onTabPress).toHaveBeenCalledTimes(1);
});
