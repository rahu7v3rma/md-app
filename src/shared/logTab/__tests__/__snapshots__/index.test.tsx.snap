// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logTab snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#FFFFFF",
      "borderRadius": 20,
      "borderWidth": 1.5,
      "elevation": 3,
      "flexDirection": "row",
      "height": 90,
      "opacity": 1,
      "shadowColor": "rgba(0,0,0,0.5)",
      "width": "44%",
    }
  }
  testID="logTab"
>
  <View
    style={
      {
        "alignItems": "center",
        "marginHorizontal": 16,
      }
    }
  >
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={21}
      bbWidth={18}
      fill="none"
      focusable={false}
      height={21}
      meetOrSlice={0}
      minX={0}
      minY={0}
      style={
        [
          {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          {
            "flex": 0,
            "height": 21,
            "width": 18,
          },
        ]
      }
      testID="logTabIcon"
      vbHeight={21}
      vbWidth={18}
      width={18}
      xml="<svg width="18" height="21" viewBox="0 0 18 21" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M11.4513 18.621C9.9292 19.477 8.0708 19.477 6.5487 18.621L3.5487 16.934C1.9743 16.048 1 14.382 1 12.576V9.424C1 7.618 1.9743 5.952 3.5487 5.066L6.5487 3.379C8.0708 2.523 9.9292 2.523 11.4513 3.379L14.4513 5.066C16.0257 5.952 17 7.618 17 9.424V11" stroke="#FF9330" stroke-width="1.5" stroke-linecap="round"/>
  <path fill-rule="evenodd" clip-rule="evenodd" d="M5.931 2.738C7.6814 1.754 9.8186 1.754 11.569 2.738L14.569 4.42599C16.3795 5.44399 17.5 7.36 17.5 9.438V11.013C17.5 11.356 17.2698 11.645 16.9556 11.735C16.9484 11.745 16.941 11.754 16.9334 11.763C16.7306 11.763 16.5159 11.765 16.2775 11.768C15.7127 11.773 15.0155 11.779 14.0312 11.763C12.7007 11.741 12.012 12.183 12.012 13.513V17.21C12.012 17.939 12.0037 18.393 11.9499 18.681C11.9343 18.926 11.7989 19.159 11.569 19.288C9.8186 20.273 7.6814 20.273 5.931 19.288L2.931 17.601C1.1205 16.582 0 14.666 0 12.589V9.43699C0 7.35999 1.1205 5.44399 2.931 4.42599L5.931 2.738ZM9 6.24999C9.4142 6.24999 9.75 6.58599 9.75 6.99999V10.701C10.1984 10.96 10.5 11.445 10.5 12C10.5 12.828 9.8284 13.5 9 13.5C8.1716 13.5 7.5 12.828 7.5 12C7.5 11.445 7.8016 10.96 8.25 10.701V6.99999C8.25 6.58599 8.5858 6.24999 9 6.24999Z" fill="#FF9330"/>
  <path d="M3 3L6 1" stroke="#FF9330" stroke-width="1.5" stroke-linecap="round"/>
  <path d="M15 3L12 1" stroke="#FF9330" stroke-width="1.5" stroke-linecap="round"/>
  <path d="M13 14.996V14.16C13 13.128 13.7861 12.702 14.7418 13.222L15.5125 13.64L16.2832 14.059C17.2389 14.578 17.2389 15.422 16.2832 15.941L15.5125 16.36L14.7418 16.778C13.7861 17.298 13 16.872 13 15.84V14.996Z" stroke="#FF9330" stroke-width="1.5" stroke-miterlimit="10" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
"
      xmlns="http://www.w3.org/2000/svg"
    >
      <RNSVGGroup
        fill={null}
        propList={
          [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M11.4513 18.621C9.9292 19.477 8.0708 19.477 6.5487 18.621L3.5487 16.934C1.9743 16.048 1 14.382 1 12.576V9.424C1 7.618 1.9743 5.952 3.5487 5.066L6.5487 3.379C8.0708 2.523 9.9292 2.523 11.4513 3.379L14.4513 5.066C16.0257 5.952 17 7.618 17 9.424V11"
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
            ]
          }
          stroke={4294939440}
          strokeLinecap={1}
          strokeWidth={1.5}
        />
        <RNSVGPath
          clipRule={0}
          d="M5.931 2.738C7.6814 1.754 9.8186 1.754 11.569 2.738L14.569 4.42599C16.3795 5.44399 17.5 7.36 17.5 9.438V11.013C17.5 11.356 17.2698 11.645 16.9556 11.735C16.9484 11.745 16.941 11.754 16.9334 11.763C16.7306 11.763 16.5159 11.765 16.2775 11.768C15.7127 11.773 15.0155 11.779 14.0312 11.763C12.7007 11.741 12.012 12.183 12.012 13.513V17.21C12.012 17.939 12.0037 18.393 11.9499 18.681C11.9343 18.926 11.7989 19.159 11.569 19.288C9.8186 20.273 7.6814 20.273 5.931 19.288L2.931 17.601C1.1205 16.582 0 14.666 0 12.589V9.43699C0 7.35999 1.1205 5.44399 2.931 4.42599L5.931 2.738ZM9 6.24999C9.4142 6.24999 9.75 6.58599 9.75 6.99999V10.701C10.1984 10.96 10.5 11.445 10.5 12C10.5 12.828 9.8284 13.5 9 13.5C8.1716 13.5 7.5 12.828 7.5 12C7.5 11.445 7.8016 10.96 8.25 10.701V6.99999C8.25 6.58599 8.5858 6.24999 9 6.24999Z"
          fill={4294939440}
          fillRule={0}
          propList={
            [
              "fill",
              "fillRule",
            ]
          }
        />
        <RNSVGPath
          d="M3 3L6 1"
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
            ]
          }
          stroke={4294939440}
          strokeLinecap={1}
          strokeWidth={1.5}
        />
        <RNSVGPath
          d="M15 3L12 1"
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
            ]
          }
          stroke={4294939440}
          strokeLinecap={1}
          strokeWidth={1.5}
        />
        <RNSVGPath
          d="M13 14.996V14.16C13 13.128 13.7861 12.702 14.7418 13.222L15.5125 13.64L16.2832 14.059C17.2389 14.578 17.2389 15.422 16.2832 15.941L15.5125 16.36L14.7418 16.778C13.7861 17.298 13 16.872 13 15.84V14.996Z"
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeLinecap",
              "strokeLinejoin",
              "strokeMiterlimit",
            ]
          }
          stroke={4294939440}
          strokeLinecap={1}
          strokeLinejoin={1}
          strokeMiterlimit={10}
          strokeWidth={1.5}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    style={{}}
    testID="logTabTitleContainer"
  >
    <Text
      adjustsFontSizeToFit={true}
      style={
        [
          {
            "fontFamily": "Poppins",
          },
          {
            "flexWrap": "wrap",
            "width": "95%",
          },
          {
            "color": "#72996B",
            "fontSize": 14,
            "fontWeight": "600",
            "lineHeight": 22,
          },
        ]
      }
      testID="logTabTitle"
    >
      Fasting Timer
    </Text>
  </View>
</View>
`;
